### ISSUE TITLE
allow to configure for where lenses are shown

### ISSUE THREAD

**Issue by jrieken** [kisstkondoros/codemetrics#3]
I understand metrics are important but the code lenses for arrow functions inside my code appear visually noisy to me. I'd prefer to see them member functions only and have a command that show the for others on demand.

**Comment by tsvetan-ganev** [comment]
The ability to toggle lenses on/off will be great. Personally, I hate every distraction inside my code while I'm editing it. I would like to see the complexity of functions, analyse the results and then hide the information while I'm actually writing code.

**Comment by kisstkondoros** [comment]
Commands for both of your needs are on the way.

@jrieken any idea how to invalidate the CodeLens cache?
smth. like 
```javascript
commands.executeCommand<vscode.CodeLens[]>('vscode.executeCodeLensProvider', vscode.window.activeTextEditor.document.uri)
```
should do it, but this fails for me with a warning message (Cannot read property 'id' of undefined) and I have no idea what causes it (I have wrong source maps I guess)

I will provide a fix which does the cache invalidation the hard way - by using edit operations... doh
CodeLensContribution onModelChange is triggered this way

**Comment by kisstkondoros** [comment]
Published the new version to the market.
I'm waiting for Your feedback :)

**Comment by jrieken** [comment]
>  any idea how to invalidate the CodeLens cache?

I am afraid to say that you cannot do that today...
